debezium-connector-oracle/src/test/resources/rest/data.sql

```diff
@@ -0,0 +1,44 @@
+-- Connect
+--CONN c##dbzuser/dbz@//localhost:1521/ORCLCDB
+CONN c##dbzuser/dbz@//localhost:1521/ORCLPDB1
```

Comment:
Could we avoid hard-coding these credentials in the start-up script?

I'm wondering if it would be less brittle of a test setup if we instead relied on the `@BeforeClass` JUnit annotation and simply got a reference to the `OracleContainer`, opened a connection directly to the database, loaded this SQL file from the file system and executed the statements as a part of our test class.

I'm just considering testing with Oracle FREE vs non-FREE containers and CDB vs non-CDB instances. wdyt?

Reply:
@Naros till me what needs to be different for other containers / instance types, then I can do the required changes.

Reply:
> please tell me what needs to be different for other containers / instance types (aka what needs to be different for non-cdb instances and what are the conditions to realize that we are in non-cdb mode)? Then I can do the required changes. Is changing the user name the only thing that I have to do? The debezium user is available on that Oracle DB test image instances?

I wish it were that simple.

So with Oracle, there are effectively two users (a) connector user and (b) test account user.  We should never use the connector user for anything because this account has to be different depending on whether the database is deployed with or without multi-tenancy.  Oracle mandates that when multi-tenancy is enabled, the user name must be prefixed with `c##` and when multi-tenancy is not it cannot be prefixed with `c##`.  This is just how Oracle works with what they call "common" vs "local" users.

So in short, you should always use the `debezium` user unless you are explicitly configuring the connector to communicate with LogMiner, XStream, or OpenLogReplicator.

The other problem is the SID or database instance you've specified.  In a multi-tenancy scenario, we would connect and work with either `ORCLPDB1` or `FREEPDB1`, depending on whether the instance is based on our images or Oracle's.  If multi-tenancy is not enabled, we'd connect to `ORCLCDB` or `FREE` , again depending on which image is being used.

This is why I recommend that you centralize this in one place by perhaps simply loading the file as a resource, iterating the lines, and executing the statements from within Java using the `OracleContainer`'s JDBC URL to connect with as this would avoid the need to use the `CONNECT` statement here and hardcode credentials.  

> For sure the code should be resilient to using different flavors of Oracle DB and I want to make those changes. But I am not a fan of doing so much in our code as there is already a mechanism in place to load such files into the database. Still, I can imagine we introduce this into the TestHelper eventually and put the SQL code from the *IT classes into the filesystem eventually. Wdyt, should I file a Jira for this?

Sure, there is a mechanism, but which makes more sense?  Duplicate the SQL files for each type of database installation with 99% of the same data but with different connection strings or actually reducing that footprint so that our technical debt and maintenance is within reason ;) 

Overall, I think we need to decide whether we want to even consider testing the REST extension with any multi-configuration setups in the first place, is there any justification for that.  If the answer is no, then picking one and sticking with that is satisfactory.  Then we simply need to evaluate what we've done primarily around the database names to be sure those are easily managed when we decide to move from Oracle 192/1 to Oracle 23 or beyond as our image base.  But if the answer is we should test varied configurations, then we have to be judicious about usernames, database names, and schema names that are used so they're mostly centralized.  TestHelper definitely does quite a bit of this already.

Reply:
Unfortunately, nothing with Oracle is ever simple, so if you want to discuss this more, feel free to schedule a time next week, and we can undoubtedly sync-up and talk about this in more detail if you think it's needed.  Anyone else on the team is welcomed to join too if they wanna ask or learn more.

Reply:
> Sure, there is a mechanism, but which makes more sense? Duplicate the SQL files for each type of database installation with 99% of the same data but with different connection strings

At the moment I am unfortunately not aware of other scenarios (for the REST extension) and their preconditions aka. how to detect them.
So I have only this one use-case in the REST Extension (mind we are in general talking about a REST endpoint that _can_ be used by the UI but in general those endpoints should provide benefit for Kafka Connect + Debezium users.

May I suggest again what I raised in the other PR:
How about introducing a new system property `<oracle.container.image.tag>${version.oracle.server}</oracle.container.image.tag>`? That is used in the maven-docker-plugin / pom.xml to select the Oracle image. In the test suite for the Oracle REST Extension I can then continue using `version.oracle.server` as the only reference to only the version number and we and QE can still customize the image used by changing that new parameter? I must say I was confused and disappointed that a variable where the name seems to only point to a version number of the oracle server is actually used to switch to another image tag.

Reply:
> How about introducing a new system property <oracle.container.image.tag>${version.oracle.server}</oracle.container.image.tag>? 
We can certainly consider this, I'm certainly not opposed to it.

> Do you have time to do a quick pair-programming-like session today? I really need to understand the differences between FREE, XS, CDB and non-CDB/PDB scenarios their conditions and differences in that Oracle DB.

Sure, just ping me and schedule something, I'm here this week and out all next week on PTO, so sooner may be ideal than waiting.

Reply:
@Naros all my changes are in now. Please have a final look. :pray: